src/nemo_safe_synthesizer/privacy/dp_transformers/dp_utils.py

```
@@ -59,6 +66,187 @@
+        logits = logits.float()
+        _log_cuda_loss_memory("after_logits_float", logits, enabled=debug_loss_memory)
+        logits = logits.view(-1, vocab_size)
+        shift_labels = shift_labels.view(-1).to(logits.device)
+        loss = loss_utils.fixed_cross_entropy(logits, shift_labels, num_items_in_batch, ignore_index, **kwargs)
+        return loss
+
+    global _CAUSAL_LM_LOSS_MEMORY_PROBE_ORIGINAL_FN, _CAUSAL_LM_LOSS_MEMORY_PROBE_PATCHED_MAPPING_KEYS
+    original = loss_utils.ForCausalLMLoss
+    _CAUSAL_LM_LOSS_MEMORY_PROBE_ORIGINAL_FN = original
+    _CAUSAL_LM_LOSS_MEMORY_PROBE_PATCHED_MAPPING_KEYS = []
+    setattr(loss_utils, "ForCausalLMLoss", probed_for_causal_lm_loss)
```

Ruff (0.15.15)

[warning] Do not call `setattr` with a constant attribute value. It is not any safer than normal property access.

Replace `setattr` with assignment

(B010)